makefiles/tools/openocd-adapters/stlink.inc.mk

```diff
@@ -8,3 +11,15 @@ ifneq (,$(DEBUG_ADAPTER_ID))
   OPENOCD_ADAPTER_INIT += -c 'hla_serial $(DEBUG_ADAPTER_ID)'
 endif
 export OPENOCD_ADAPTER_INIT
+
+# select OpenOCD configuration if not already set by the board configuration
+ifeq (,$(OPENOCD_CONFIG))
+  OPENOCD_BOARD_CFG = $(RIOTBOARD)/$(BOARD)/dist/openocd.cfg
+  ifeq "$(OPENOCD_BOARD_CFG)" "$(wildcard $(OPENOCD_BOARD_CFG))"
```

Comment:
You do not need to set `OPENOCD_CONFIG` when it is the default value as it will be set in `openocd.sh`

So something like this would work I think.

``` diff
diff --git a/makefiles/tools/openocd-adapters/stlink.inc.mk b/makefiles/tools/openocd-adapters/stlink.inc.mk
index 6e7fdc6..0a5e008 100644
--- a/makefiles/tools/openocd-adapters/stlink.inc.mk
+++ b/makefiles/tools/openocd-adapters/stlink.inc.mk
@@ -14,11 +14,8 @@ export OPENOCD_ADAPTER_INIT
 
 # select OpenOCD configuration if not already set by the board configuration
 ifeq (,$(OPENOCD_CONFIG))
-  OPENOCD_BOARD_CFG = $(RIOTBOARD)/$(BOARD)/dist/openocd.cfg
-  ifeq "$(OPENOCD_BOARD_CFG)" "$(wildcard $(OPENOCD_BOARD_CFG))"
-    # Use the openocd config file in the board directory if it exists
-    export OPENOCD_CONFIG := $(RIOTBOARD)/$(BOARD)/dist/openocd.cfg
-  else
+  # select OpenOCD configuration if there is none in the default location
+  ifeq (,$(wildcard $(RIOTBOARD)/$(BOARD)/dist/openocd.cfg))
     # select OpenOCD configuration depending on CPU type
     export OPENOCD_CONFIG := $(RIOTBASE)/boards/common/st/configs/$(CPU).cfg
   endif
```

I checked with this both in master and this pr with the diff.
And checked that for `stm32f769i-disco` it is the default file in `openocd.sh`.
``` bash
$ make --no-print-directory -C examples/hello-world/ BOARD=stm32f769i-disco info-debug-variable-OPENOCD_CONFIG

$ make --no-print-directory -C examples/hello-world/ BOARD=nucleo-l152re info-debug-variable-OPENOCD_CONFIG
/home/harter/work/git/RIOT/boards/common/st/configs/stm32l1.cfg
```

Reply:
This is what I did initially but for some reason it was not working: the `ifeq` was always false and the common configuration always used. I tested that with a nucleo-l073rz and b-l072z-blabla

Reply:
It works on my computer.

With the diff:

```
make --no-print-directory -C examples/hello-world/ BOARD=nucleo-l073rz info-debug-variable-OPENOCD_CONFIG
/home/harter/work/git/RIOT/boards/common/st/configs/stm32l0.cfg
```

For b-l072-lrwan-1 it is using the one in dist/openocd.cfg
```
make --no-print-directory -C examples/hello-world/ BOARD=b-l072z-lrwan1 info-debug-variable-OPENOCD_CONFIG

# Which is wanted as it has a different configuration
diff -u boards/b-l072z-lrwan1/dist/openocd.cfg boards/common/st/configs/stm32l0.cfg
--- boards/b-l072z-lrwan1/dist/openocd.cfg	2018-05-14 15:15:07.324865950 +0200
+++ boards/common/st/configs/stm32l0.cfg	2018-05-29 22:49:54.915624904 +0200
@@ -1,3 +1,3 @@
 source [find target/stm32l0.cfg]
-
-reset_config trst_and_srst srst_nogate connect_assert_srst
+reset_config srst_only
+$_TARGETNAME configure -rtos auto
```

Reply:
If there really is a whitespace non-robustness problem it can be done with `words` also

`ifeq (0,$(words $(wildcard $(RIOTBOARD)/$(BOARD)/dist/openocd.cfg)))`

Reply:
Using make words function does the trick.